File: genaibackend/src/main/java/com/clothingstore/entity/Coupon.java
package com.clothingstore.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "coupons")
public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(unique = true, nullable = false)
    private String code;
    
    @Column(nullable = false)
    private BigDecimal discountPercent;
    
    @Enumerated(EnumType.STRING)
    private CouponStatus status = CouponStatus.ACTIVE;
    
    @Column(name = "created_date")
    private LocalDateTime createdDate;
    
    @PrePersist
    protected void onCreate() {
        createdDate = LocalDateTime.now();
    }
    
    public enum CouponStatus {
        ACTIVE, EXPIRED, USED
    }
    
    // Constructors
    public Coupon() {}
    
    public Coupon(String code, BigDecimal discountPercent) {
        this.code = code;
        this.discountPercent = discountPercent;
    }
    
    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public String getCode() { return code; }
    public void setCode(String code) { this.code = code; }
    
    public BigDecimal getDiscountPercent() { return discountPercent; }
    public void setDiscountPercent(BigDecimal discountPercent) { this.discountPercent = discountPercent; }
    
    public CouponStatus getStatus() { return status; }
    public void setStatus(CouponStatus status) { this.status = status; }
    
    public LocalDateTime getCreatedDate() { return createdDate; }
    public void setCreatedDate(LocalDateTime createdDate) { this.createdDate = createdDate; }
}

File: backend-dotnet/Models/Coupon.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClothingStore.Models
{
    [Table("coupons")]
    public class Coupon
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        
        [Required]
        [Column("code")]
        public string Code { get; set; } = string.Empty;
        
        [Required]
        [Column("discount_percent", TypeName = "decimal(5,2)")]
        public decimal DiscountPercent { get; set; }
        
        [Column("status")]
        public string Status { get; set; } = "ACTIVE";
        
        [Column("created_date")]
        public DateTime CreatedDate { get; set; } = DateTime.Now;
    }
}